lms/djangoapps/monitoring/middleware.py

```diff
@@ -30,11 +29,6 @@ def __call__(self, request):
         response = self.get_response(request)
         return response
 
-    def process_view(self, request, view_func, view_args, view_kwargs):
-        self._set_view_func_compare_metric(view_func)
```

Comment:
Before PR #24184 this relied on `process_view`, rather than calling the function from `__call__` -- any reason not to restore it to that pattern? (I don't know much about Django middleware, just curious about the diff.)

Reply:
(I see now that this and the previous change are in fact *aimed at* getting rid of `process_view`; I don't understand why that was wanted, but the change looks fine!)

Reply:
* Django has old style Middleware with `process_request`, `process_response`, etc. that you will see all over Open edX.  They mostly use `MiddlewareMixin` which makes the old-style compatible with the new style.  This Middleware uses the new style.
* Both in old and new style, `process_view` is a separate method that is called later in the lifecycle after `process_request` (or new style equivalent) is called for all the Middleware.  We moved the mapping earlier (process request) in order to not miss some Transactions that were never making it as for as `process_view`.

Reply:
Ah, thanks!

Might be worth including that last bit in the commit message when merging/squashing.

Reply:
I updated the commit message so I won't forget tomorrow.  I didn't include as much info about new/old style middleware, but explained the previous refactor.